Mathlib/Topology/Algebra/Valued/LocallyCompact.lean

```diff
@@ -171,118 +170,132 @@ section CompactDVR
-  simp only [Subtype.mk_lt_mk] at hm
-  -- well, it is in the ring, so there is something in the cover that covers it,
-  -- and it must be a sphere since it is larger than the threshold element by virtue of
-  -- `v x < v l < v m`.
-  obtain ⟨n, hn, hn'⟩ : ∃ n ∈ t, v (n : K) = v m := by
-    refine htm ⟨m, hm.right.le⟩ (hxw.trans (hm.left.trans_le' ?_))
-    rw [hvl]
-    refine Finset.le_max' _ _ ?_
-    simp only [Finset.mem_image, Finset.mem_filter, Function.comp_apply]
-    exact ⟨w, ⟨hwt, hw1⟩, rfl⟩
-  rw [← hn'] at hm -- clean up what valuations we refer to
-  -- to supply the contradiction, we have `v l < v n`, now prove that also `v n ≤ v l`
-  refine hm.left.not_ge ?_
-  -- which is the case since `‖l‖ = u.max' ..` and the property of `Finset.max'`
-  rw [hvl]
-  refine Finset.le_max' _ _ ?_
-  simp only [Finset.mem_image, Finset.mem_filter, Function.comp_apply]
-  use n
-  simp [hn, hm.right]
-
-lemma isDiscreteValuationRing_of_compactSpace [hv : (Valued.v : Valuation K Γ₀).RankOne]
+  obtain ⟨t, ht⟩ := this
+  classical
+  refine (t.finite_toSet.dependent_image ?_).subset ?_
+  · refine fun i hi ↦ if hi' : v i ≤ z then z else Units.mk0 ⟨(v i), by simp⟩ ?_
+    push_neg at hi'
+    exact Subtype.coe_injective.ne_iff.mp (hi'.trans' z0).ne'
+  · intro i
+    simp only [Set.mem_Icc, Finset.mem_coe, exists_prop, Set.mem_setOf_eq, and_imp]
+    -- we get the `c` from the cover that covers our arbitrary `i` with its set
+    obtain ⟨c, hc⟩ := i.val.prop
+    intro hzi hi1
+    have hj := ht (hc.trans_le hi1)
+    simp only [Set.mem_iUnion, exists_prop, U] at hj
+    obtain ⟨j, hj, hj'⟩ := hj
+    use j, hj
+    -- and this `c` is either less than or greater than (or equal to) the threshold element
+    split_ifs at hj' with hcj
+    · simp only [Set.mem_setOf_eq, hc, Subtype.coe_le_coe, Units.val_le_val] at hj'
+      simp [hcj, le_antisymm hj' hzi]
+    · simp only [Set.mem_setOf_eq] at hj'
+      rw [dif_neg hcj]
+      simp [← hj', hc]
+
+lemma mulArchimedean_mrange_of_isCompact_integer (hc : IsCompact (X := K) 𝒪[K]) :
+    MulArchimedean (MonoidHom.mrange (Valued.v : Valuation K Γ₀)) := by
+  rw [← Units.mulArchimedean_iff]
+  obtain ⟨_⟩ := locallyFiniteOrder_units_mrange_of_isCompact_integer hc
+  exact MulArchimedean.of_locallyFiniteOrder
+
+lemma isPrincipalIdealRing_of_compactSpace [hc : CompactSpace 𝒪[K]] :
+    IsPrincipalIdealRing 𝒪[K] := by
+  -- TODO: generalize to `Valuation.Integer`, which will require showing that `IsCompact`
+  -- pulls back across `TopologicalSpace.induced` from a `LocallyCompactSpace`.
+  -- The strategy to show that we have a PIR is by contradiction,
+  -- assuming that the range of the valuation is densely ordered.
+  -- We can also construct that it has a locally finite ordered, by compactness
+  -- which leads to a contradiction.
+  -- the key result is that a valuation ring that maps into a `MulArchimedean` value group
+  -- is a PIR iff the value group is not densely ordered.
```

Comment:
Where does this TODO end? Is it all a TODO? My impression is that some of this para is an explanation of the proof which is about to happen but the first part is saying what should be done later. Can you add in a newline somewhere maybe?

Reply:
And I think this should be moved to `locallyFiniteOrder_units_mrange_of_isCompact_integer` instead.